util/settings.lua

```diff
@@ -45,14 +45,63 @@ mail.settings = {
         type = "string", default = "%Y-%m-%d %X", group = "other", index = 3, label = S("Date format"),
         dataset = {"%Y-%m-%d %X", "%d/%m/%y %X", "%A %d %B %Y %X"}, format = os.date
     },
+    mute_list = {
+        type = "list", default = {}, group = "spam", index = 1,
+        label = S("Mute list")
+    },
 }
 
 mail.settings_groups = {
     { name = "notifications", label = S("Notifications")},
     { name = "message_list",  label = S("Message list")},
+    { name = "spam",          label = S("Spam")},
     { name = "other",         label = S("Other")}
 }
 
-for s, _ in pairs(mail.settings) do
+for s, d in pairs(mail.settings) do
 	mail.selected_idxs[s] = {}
+	if d.type == "list" then
+        mail.selected_idxs["index_" .. s] = {}
+    end
+end
+
+function mail.settings.mute_list.check(name, value)
+    local valid_players = {}
+    for _, p in ipairs(value) do
+        if p ~= name and minetest.player_exists(p) then
+            table.insert(valid_players, p)
+        end
+    end
+    return valid_players
+end
+
+--[[function mail.settings.mute_list.sync(name)
```

Comment:
what are you planning to do with this function, this looks like a cpu-hog :thinking:

Reply:
`check` (the end of the part of code) or `sync` ?
- `check` is to be sure all usernames given are existing
- `sync` is to get data from beerchat, but it is temporary disabled waiting for mute fixes merged

Reply:
I do still think that data `sync` for muting could be a way better as event based specific checks than a static lists and settings sync.

Additionally for beerchat muting to stay sync globally both ways you do not really have to transfer any static data at all and also already has fairly good performance for most parts when used directly.

So making this side also event based would allow for example (if something in this list does not make sense feel free to ask for clarification):
* getting rid of somewhat complicated data transfer between the mods.
* likely better performance without additional optimizations, caching or usage restrictions.
* no loading or syncing of data for players who do not use `/mail`.
* data only fetched when it is actually needed.
* data always up to date, no synchronization required ever.
* no merging of data and no conflict resolutions for merge conflicts.
* does not have to worry about merging possible offline data updates (external systems for example).
* overall simpler compared to data synchronization.
*  automatic handling both public list (other mods) and private list (mail mod), just user interface could use specific code for public/private distinction.

I've not put a lot thought on config sync system so I'd love to hear what problems it can solve and why it would be preferred over on-demand API calls or other event based checks.

Reply:
Non-using `beerchat` disables the `sync`ing script. This is on purpose : it syncs list only when the setting is called. I don't want to have mail storage outside mail : transfer is better than putting muting lists in beerchat. If beerchat is deleted, everything is out.

Reply:
> If beerchat is deleted, everything is out.

Did I get it right: basically you want that beerchat configrations keep affecting mail even after beerchat has been removed?

Reply:
:+1: Yes, and it saves resources by syncing only when needed.

Reply:
It actually very likely wastes resources and has worse performance as it syncs what potentially isn't needed but also it is probably simplest way to keep configurations (both wanted and unwanted) after removing source mod.

Reply:
> It actually very likely wastes resources and has worse performance as it syncs what potentially isn't needed but also it is probably simplest way to keep configurations (both wanted and unwanted) after removing source mod.

If you have a better idea tell me, likely beerchat could have a similar `transfer` function ?

Reply:
> > It actually very likely wastes resources and has worse performance as it syncs what potentially isn't needed but also it is probably simplest way to keep configurations (both wanted and unwanted) after removing source mod.
> 
> If you have a better idea tell me, likely beerchat could have a similar `transfer` function ?

See comment in another thread (code comments) https://github.com/mt-mods/mail/pull/128/files#r1451568438